distributed/tests/test_client.py

```diff
@@ -6461,24 +6461,15 @@ async def f(stacklevel, mode=None):
     assert "cdn.bokeh.org" in data
 
 
-@gen_cluster(nthreads=[])
-async def test_client_gather_semaphore_loop(s):
-    async with Client(s.address, asynchronous=True) as c:
-        assert c._gather_semaphore._loop is c.loop.asyncio_loop
-
-
 @gen_cluster(client=True)
 async def test_as_completed_condition_loop(c, s, a, b):
     seq = c.map(inc, range(5))
     ac = as_completed(seq)
+    async for _ in ac:
+        pass
```

Comment:
What is the purpose of this change?

Reply:
on python 3.10 async resources like asyncio.Condition bind the loop late: https://github.com/python/cpython/blob/ba76f901923d80ad9b24bb1636aa751d55e0c768/Lib/asyncio/mixins.py#L28

by iterating `ac` we're interacting with the `ac.condition` so it gets bound to a loop

Reply:
Maybe worth including a comment then?